src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java

```diff
@@ -364,7 +373,7 @@ public boolean shouldConnect() {
       tryGetProject() != null &&
       !autoServiceAccountUnavailable();
```

Comment:
Add `!containsMacro(NAME_SERVICE_ACCOUNT_JSON)` in this method.

Reply:
Fixed